strings/my_vsnprintf.c

```diff
@@ -723,20 +744,39 @@ size_t my_vsnprintf_ex(CHARSET_INFO *cs, char *to, size_t n,
 
     fmt= check_longlong(fmt, &have_longlong);
 
-    if (*fmt == 's' || *fmt == 'T')			/* String parameter */
+    switch(*fmt) {
+    case 's':
+    case 'T':
+    case 'b':
     {
+      /* String parameter */
       reg2 char *par= va_arg(ap, char *);
-      to= process_str_arg(cs, to, end, (longlong) length, width, par,
-                          print_type, (*fmt == 'T'));
-      continue;
-    }
-    else if (*fmt == 'b')				/* Buffer parameter */
-    {
-      char *par = va_arg(ap, char *);
-      to= process_bin_arg(to, end, width, par);
+      my_bool suffix_b= *fmt == 'b', suffix_t= *fmt == 'T';
+      switch (fmt[1]) // look-ahead
```

Comment:
Have we done a check that there is a character to look ahead to?

Reply:
No. I assumed that, because `fmt` is a ␀-terminated char sequence, knowing that `fmt[0]` (`*fmt`) is not ␀, `fmt[1]` would be either a char in the sequence or the terminating ␀.